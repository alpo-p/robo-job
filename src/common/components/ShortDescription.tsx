import { useTheme } from '@react-navigation/native'
import React from 'react'
import { TouchableOpacity, View } from 'react-native'

import ContentText from './ContentText'
import RoboText from './RoboText'

interface Props {
  description: string
  useWiderStyle?: boolean
  navigateToDetails?: () => void
}

const Description: React.FC<Props> = ({
  description,
  useWiderStyle,
  navigateToDetails,
}) => {
  const { colors } = useTheme()
  return (
    <View
      style={{
        width: useWiderStyle ? '100%' : 280,
      }}
    >
      <ContentText>{description}</ContentText>
      {navigateToDetails && (
        <TouchableOpacity onPress={navigateToDetails} style={{ marginTop: 8 }}>
          <RoboText color={colors.primary} weight="bold">
            Read more
          </RoboText>
        </TouchableOpacity>
      )}
    </View>
  )
}

export default Description
