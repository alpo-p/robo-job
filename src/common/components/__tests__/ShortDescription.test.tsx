import React from 'react'
import rendererCreateWithNavigation from '../../../testUtils/rendererCreateWithNavigation'
import ShortDescription from '../ShortDescription'

describe('ShortDescription', () => {
  it('should match snapshot', () => {
    const tree = rendererCreateWithNavigation(
      <ShortDescription description="testi" />,
    )
    expect(tree).toMatchSnapshot()
  })
})
