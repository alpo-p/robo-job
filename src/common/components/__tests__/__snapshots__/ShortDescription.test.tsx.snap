// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ShortDescription should match snapshot 1`] = `
<View
  style={
    Object {
      "width": 280,
    }
  }
>
  <Text
    style={
      Object {
        "color": "#000000",
        "fontFamily": "DamascusLight",
        "fontSize": 14,
        "lineHeight": 24,
        "textAlign": "justify",
        "textTransform": "none",
      }
    }
  >
    testi
  </Text>
</View>
`;
